thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java

```diff
@@ -217,73 +218,123 @@ public Response computeSeverity(@NotNull @QueryParam("collection") String collec
    * Regenerates anomalies for each window separately
    *
    * As the anomaly result regeneration is a heavy job, we move the function from Dashboard to worker
-   * @param id anomaly function id
+   * @param ids a string containing multiple anomaly function ids, separated by comma (e.g. f1,f2,f3)
    * @param startTimeIso The start time of the monitoring window (in ISO Format), ex: 2016-5-23T00:00:00Z
    * @param endTimeIso The start time of the monitoring window (in ISO Format)
    * @param isForceBackfill false to resume backfill from the latest left off
+   * @param speedup whether this backfill should speedup with 7-day window
    * @return HTTP response of this request
    * @throws Exception
    */
   @POST
-  @Path("/{id}/replay")
-  public Response generateAnomaliesInRange(@PathParam("id") @NotNull String id,
+  @Path("/{ids}/replay")
+  public Response generateAnomaliesInRange(@PathParam("ids") @NotNull String ids,
       @QueryParam("start") @NotNull String startTimeIso,
       @QueryParam("end") @NotNull String endTimeIso,
-      @QueryParam("force") @DefaultValue("false") String isForceBackfill) throws Exception {
-    final long functionId = Long.valueOf(id);
+      @QueryParam("force") @DefaultValue("false") String isForceBackfill,
+      @QueryParam("speedup") @DefaultValue("false") final Boolean speedup) throws Exception {
     final boolean forceBackfill = Boolean.valueOf(isForceBackfill);
-    AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(functionId);
-    if (anomalyFunction == null) {
+    final List<Long> functionIdList = new ArrayList<>();
+    for (String functionId : ids.split(",")) {
+      AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(Long.valueOf(functionId));
+      if (anomalyFunction != null && anomalyFunction.getIsActive()) {
+        functionIdList.add(Long.valueOf(functionId));
+      } else {
+        // Check if the anomaly function is active
+        if (!anomalyFunction.getIsActive()) {
```

Comment:
should you check anomalyFunction == null here first? Also this treatment of different function ids together feels very arbitrary.